windows/README.md

```diff
@@ -26,7 +26,7 @@ tidal login
     - Specify `-UserName` parameter value
     - Store the password securely by running `.\save_password.ps1` script
 
-4) Ensure you have a file that has a list of all the hostnames you want to scan. By default, Machine Stats looks for `servers.txt`, but you can also specify any custom location with `-ServersPath` parameter. The hosts will need to be accessible via your network connection from the machine that you run this from.
+4) Ensure you have a file that has a list of all the private IP addresses you want to scan. For analyzing machines over the internet, you can use the public IPv4 DNS name. By default, Machine Stats looks for `servers.txt`, but you can also specify any custom location with `-ServersPath` parameter. The hosts will need to be accessible via your network connection from the machine that you run this from.
```

Comment:
I think we should also add an example file here, same as `hosts.example` in UNIX implementation and also on the guides (or mention in the guides that the example file is within the code base).

Reply:
That is a good idea @pranavmalaviya. 👍

Reply:
good work Sam. very interesting, do we know why it doesn't work with hostnames? Is it because in this case, the hostnames do not resolve to IPs on the server that is running machine-stats? I see it does work with hostnames defined via DNS (ie 'the public IPv4 DNS name').

`the public IPv4 DNS name` - this is a very ec2 AWS-centric term. it means 'the fully qualified domain name (FQDN) for this host, that resolves to a publically routable IPv4 IP address.' - most users aren't connecting to ec2 instances for this use case here. The term isn't commonly used or understood outside the context of ec2.


It should be that machine-stats is able to connect any server via an IP address provided that is routable from where it's being run (could be private or public). And any server via a hostname that is can be resolved via DNS from where machine stats in being run. (Again DNS resolution could be public or private). I believe it's the [Invoke-Command and -ComputerName parameter](https://docs.microsoft.com/en-us/powershell/module/microsoft.powershell.core/invoke-command?view=powershell-7.2#parameters)

I see a potential source of confusion, `hostnames` - are not necessarily resolvable (to IPs), but often are. For example, from memory here, the 'hostname' that AWS gives ec2 instances, is not an FQDN in public DNS. they assign it a different DNS entry. - this varies a lot from on-premise enterprise setups, where hostnames are often resolvable on local networks.

One suggestion on how we can improve the docs here is phrasing like this:

>Ensure you have a file that has a list of hosts to be scanned. Hosts can be specified either as IP addresses or as hostnames that resolve via DNS. In either case, the hostnames and IP addresses must be resolvable (private or global DNS) and routable (either locally or over the internet) from the machine that machine-stats is running on.

Reply:
Excellent points Phil, thanks for the feedback. Networking is not an area I have much experience in - it seems that the core issue is there's no DNS resolution for hostnames in the same aws subnet. I think that's a much better explanation for the readme.